### DISCUSSION TITLE
Error encountered at Section 3 Lesson 16 when using the 'make deploy' in terminal in VScode

### DISCUSSION THREAD

**Discussion by Taffy91** [Cyfrin/security-and-auditing-full-course-s23#91] [Q&A]
At lesson 16 of Section 3, named 'Write an amazing finding: Proof of code', I ran anvil to make a fake blockchain running as instructed by Patrick. I created a new shell to deploy the PasswordStore to the locally running blockchain. I then ran the DeployPassword script which was found in the 'Makefile'. I ran the 'make deploy' command in the visual studio code terminal to deploy 'PasswordStore' to the address copied on the locally running anvil chain because Patrick said it was time to do so. This was @0.32-47s of this lesson. Upon doing so I got an error that looks like this in my visual studio code terminal, 
 'Error: 
solc exited with signal: 9 (SIGKILL)
<empty output>
make: *** [Makefile:45: deploy] Error 1'

How do I get past this error. I just want to say thanks in advance to the person or persons that would help in this regard.

**Comment by PatrickAlphaC** [comment]
Hello! Thanks for making this issue. 

Could you try to git clone the repo again? I'm not sure I have enough info to triage this. Have you taken the foundry full course first?

**Reply by cromewar** [reply]
Please, @Taffy91 give us a little bit more context on this one. Please check [how to ask questions](https://github.com/Cyfrin/foundry-full-course-f23/blob/main/how-to-ask-a-question.md).

**Reply by Taffy91** [reply]
@cromewar & @PatrickAlphaC  good day guys I apologize for not giving more context. I didn't realise it wasn't enough. I tried what @PatrickAlphaC said to do and it worked. I apologize for the late reply also, I have been very busy where I am currently. I'll try to reply quicker and give more context if I have any more questions in the future. I genuinely thought that it was enough info. But thanks a lot @PatrickAlphaC for the advice it literally worked as I did what you advised. I was doing some other work why I didn't reply in a quicker fashion.

**Reply by Taffy91** [reply]
@cromewar I read what you suggested and updated the question that I asked. I hope even though my situation was resolved that the way I updated my question made sense. Please tell me if this was ant clearer please, I tried to be more verbose.

**Comment by akmalzufaaaf** [comment]
Hi @Taffy91,

For anyone facing the SIGKILL (Out of Memory) error—or even weird dependency issues—during make deploy, I found a reliable workaround.

The Fix:
Don't let `make` handle the compilation. Do it manually first to reduce the load.
1. Run `forge build` explicitly.
2. Then run `make deploy`.

Why:
The `make deploy` command in the Makefile combines compilation and execution. Splitting them up isolates the processes and prevents the compiler from getting killed by the OS (which is what SIGKILL implies).

<img width="716" height="202" alt="image" src="https://github.com/user-attachments/assets/0b95aaba-f3c7-4bb2-bd73-483fad0f5fd8" />

It worked for me to stabilize the deployment process. Hope it helps!

**Reply by EngrPips** [reply]
Thanks for this heads-up.

**Reply by akmalzufaaaf** [reply]
Hope it helps!
